mkt/inapp/views.py

```diff
@@ -52,9 +52,13 @@ def get_queryset(self):
         return InAppProduct.objects.filter(webapp=self.get_app())
 
     def get_app(self):
+        origin = self.kwargs['origin']
         if not hasattr(self, 'app'):
-            self.app = get_object_or_404(Webapp,
-                                         app_domain=self.kwargs['origin'])
+            if origin.startswith('marketplace:'):
+                lookup = dict(guid=origin.replace('marketplace:', ''))
```

Comment:
I feel like these should use `{'guid': origin.replace('marketplace:', '')}` instead of calling `dict` but I don't know if that's explicitly in our style guide.